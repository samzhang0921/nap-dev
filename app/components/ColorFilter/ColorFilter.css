.ColorFilter {
  composes: catFilterList from '../CategoryFilter/CategoryFilter.css';
}

h4 {
  font-weight: 400;
  font-family: GillSans, "Gill Sans MT", Calibri, sans-serif;
  text-transform: uppercase;
  display: inline-block;
  margin-top: 0;
  margin-bottom: 11px;
}

.clearFilter {
  composes: clearFilter from '../DesignerFilter/DesignerFilter.css';
}

.filterCheckbox {
composes: filterCheckbox from '../Designer/Designer.css';
}
