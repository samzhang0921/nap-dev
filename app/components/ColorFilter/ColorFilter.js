import React, {Component} from 'react';
import style from './ColorFilter.css';

const ColorFilter = props => {
  return (
    <div className={style.ColorFilter}>
    <h4>COLOR</h4>
    <a className={style.clearFilter} href="?">Clear</a>
    <ul>
      <li data-colour-id="1001" className="not-selected">
        <span className="filter-checkbox"></span>
        <span className="filter-name">Animal print</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="2" className="not-selected">
        <span className="filter-checkbox"></span>
        <span className="filter-name">Black</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="3" className="not-selected">
        <span className="filter-checkbox"></span>
        <span className="filter-name">Blue</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="6" className="not-selected">
        <span className="filter-checkbox"></span>
        <span className="filter-name">Brown</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="7" className="not-selected">
        <span className="filter-checkbox"></span>
        <span className="filter-name">Burgundy</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="17" className="not-selected">
          <span className="filter-checkbox"></span>
          <span className="filter-name">Gold</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="18" className="not-selected">
        <span className="filter-checkbox"></span>
        <span className="filter-name">Gray</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="19" className="not-selected">
          <span className="filter-checkbox"></span>
          <span className="filter-name">Green</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="104" className="not-selected">
        <span className="filter-checkbox"></span>
        <span className="filter-name">Metallic</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="1002" className="not-selected">
        <span className="filter-checkbox"></span>
        <span className="filter-name">Neutrals</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="31" className="not-selected">
          <span className="filter-checkbox"></span>
          <span className="filter-name">Orange</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="33" className="not-selected">
        <span className="filter-checkbox"></span>
        <span className="filter-name">Pink</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="36" className="not-selected">
        <span className="filter-checkbox"></span>
        <span className="filter-name">Purple</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="37" className="not-selected">
        <span className="filter-checkbox"></span>
        <span className="filter-name">Red</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="69" className="not-selected">
        <span className="filter-checkbox"></span>
        <span className="filter-name">Silver</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="43" className="not-selected">
        <span className="filter-checkbox"></span>
        <span className="filter-name">White</span>
        <div className="clear"></div>
      </li>

      <li data-colour-id="44" className="not-selected">
        <span className="filter-checkbox"></span>
        <span className="filter-name">Yellow</span>
        <div className="clear"></div>
      </li>
    </ul>
    </div>
  )
}

export default ColorFilter;
