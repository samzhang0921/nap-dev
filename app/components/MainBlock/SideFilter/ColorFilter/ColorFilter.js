import React from 'react';
import {render} from 'react-dom';
import style from './ColorFilter.css';
import Color from '../Color/Color';
export default class ColorFilter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
        clearClicked:false
      };
      this.clearColor = this.clearColor.bind(this);
    }

    clearColor () {
      const {clearClicked} = this.state;
      this.setState({
          clearClicked:!clearClicked
      });
      this.props.updateColorList(0);
    }
    componentWillUpdate(nextProps, nextState){
      if (this.props.categoryRefresh !== nextProps.categoryRefresh){
        this.setState({
          clearClicked:!this.state.clearClicked
        })
      }
    }
render() {
    if(!this.props.colorList) {return <div className={style.ColorFilter} >Loading</div>};
    return (
      <div className={style.ColorFilter}>
        <h4> COLOR </h4>
        <a className = {style.clearFilter} onClick={()=>this.clearColor()}>Clear</a>
        <ul>
        {this.props.colorList.map((pColor) => {
          return <Color
          key = {pColor.id}
          id = {pColor.id}
          name = {pColor.name}
          updateColorList = {this.props.updateColorList}
          categoryRefresh={this.props.categoryRefresh}
          clearClicked = {this.state.clearClicked}
          />
        })}
        </ul>
      </div>
    )
  }
}
