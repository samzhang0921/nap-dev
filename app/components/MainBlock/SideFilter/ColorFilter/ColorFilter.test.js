import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import ColorFilter from './ColorFilter';

describe('ColorFilter', () => {
  const pColor =  {
"id": 3,
"name": "Blue"
};

  it('should mount item ', () => {
    const tree = toJson(mount(<ColorFilter data={[]}/>));
    expect(tree).toMatchSnapshot();
  });

  it('should match its snapshot with items', () => {

    const tree = renderer.create(<ColorFilter colorList={[pColor]}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should change class on the onClick function', () => {
    const onChange = jest.fn();
    
      const component = mount(<ColorFilter colorList={[pColor]} updateColorList={onChange}/>);
      component.find('.clearFilter').simulate('click');
      expect(component.find('.filterCheckedbox')).toHaveLength(0);
    });
});
