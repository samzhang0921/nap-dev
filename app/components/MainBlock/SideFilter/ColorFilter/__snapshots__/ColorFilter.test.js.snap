// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorFilter should match its snapshot with items 1`] = `
<div
  className="ColorFilter"
>
  <h4>
     COLOR 
  </h4>
  <a
    className="clearFilter"
    onClick={[Function]}
  >
    Clear
  </a>
  <ul>
    <li
      data-category-id={3}
      onClick={[Function]}
    >
      <span
        className="colorCheckbox"
      />
      <span
        className="colorName"
      >
        Blue
      </span>
      <div
        className="clear"
      />
    </li>
  </ul>
</div>
`;

exports[`ColorFilter should mount item  1`] = `
<ColorFilter
  data={Array []}
>
  <div
    className="ColorFilter"
  >
    Loading
  </div>
</ColorFilter>
`;
